controlplane/api/v1alpha1/zz_generated.conversion.go

```diff
@@ -423,6 +423,8 @@ func autoConvert_v1beta1_RKE2ControlPlaneSpec_To_v1alpha1_RKE2ControlPlaneSpec(i
 		return err
 	}
 	out.Replicas = (*int32)(unsafe.Pointer(in.Replicas))
+	// WARNING: in.Version requires manual conversion: does not exist in peer-type
```

Comment:
This means that version needs to be stored as annotation, and restored manually, if for some reason the RKE2 provider version is downgraded. Maybe needs addressing.

Reply:
There is a manual conversion [here](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/43f5e60485bf2320de7699ad4e6f21d791b841da/controlplane/api/v1alpha1/conversion.go#L45) that handles the new fields `Version` and `MachineTemplate`.